lib/src/git.rs

```diff
@@ -213,28 +213,54 @@ pub enum GitExportError {
     InternalGitError(#[from] git2::Error),
 }
 
-/// Reflect changes made in the Jujutsu repo compared to our current view of the
-/// Git repo in `mut_repo.view().git_refs()`. Returns a list of names of
-/// branches that failed to export.
-// TODO: Also indicate why we failed to export these branches
+/// Export changes to branches made in the Jujutsu repo compared to the Git
+/// repo. Returns a list of names of branches that failed to export.
+///
+/// We ignore branches that are conflicted (were changed in the Git repo
+/// compared to our last remembered view of the Git repo stored on Git). These
+/// will be marked conflicted by the next `jj git import`.
+///
+/// We do not export tags and other refs at the moment, since these aren't
+/// supposed to be modified by JJ. For them, the Git state is considered
+/// authoritative.
+//
+// TODO: Also indicate the reason for any branches we failed to export
 pub fn export_refs(
     mut_repo: &mut MutableRepo,
     git_repo: &git2::Repository,
 ) -> Result<Vec<String>, GitExportError> {
-    // First find the changes we want need to make without modifying mut_repo
+    // It is important for `last_seen_view` to be unaffected by `jj undo`.
+    // Short-term TODO: this comment is expanded upon in a follow-up commit
+    let last_export_path = mut_repo.base_repo().repo_path().join("git_last_seen_refs");
+    let last_seen_view = crate::simple_op_store::read_view_from_file(last_export_path.clone())
+        .unwrap_or(crate::op_store::View::default());
+    let (new_exported_view, failed_branches) =
+        export_branches_unchanged_since_old_view(mut_repo, &View::new(last_seen_view), git_repo)?;
+    crate::simple_op_store::write_view_to_file(new_exported_view.store_view(), last_export_path)
+        .map_err(|err| GitExportError::StateReadWriteError(err.to_string()))?;
+    Ok(failed_branches)
+}
+
+/// Exports unconflicted branches.
+///
+/// Potential conflicts between the Git repo state and the JJ state are detected
+/// by using `old_view` as the conflict base. Returns the list of unexported
+/// branches and the new git state (with successfully exported branches moved).
+fn export_branches_unchanged_since_old_view(
```

Comment:
I think I understand what you mean by "unchanged" here, but I also find it confusing because we export only branches that have changed in jj compared to the old view. I think it's best to just drop "unchanged" in the name. The function exports branches that have changed between `old_view` and `mut_repo.view()`, so I think `export_branches_since_old_view()` makes sense. I think your "unchanged" was referring to the fact that it tries to not overwrite changes made in the git repo since our last seen state, but I think it's enough to just explain that in the comment, as you already have.